armi/reactor/converters/uniformMesh.py

```diff
@@ -329,8 +568,62 @@ def plotConvertedReactor(self):
                 showBlockAxMesh=True,
             )
 
+    def reset(self):
+        """Clear out stored attributes and reset the global assembly number."""
+        self.reactorParamNames = []
+        self.blockScalarParamNames = []
+        self.blockArrayParamNames = []
+        self._cachedReactorCoreParamData = {}
+        super().reset()
+
     def _setParamsToUpdate(self):
-        """Gather a list of parameters that will be mapped between reactors."""
+        """Activate conversion of various paramters."""
+        self.reactorParamNames = []
+        self.blockScalarParamNames = []
+        self.blockArrayParamNames = []
+
+    def _checkConversion(self):
+        """Perform checks to ensure conversion occurred properly."""
+        pass
+
+    def _computeAverageAxialMesh(self):
+        """Computes an average axial mesh based on the core's reference assembly."""
+        src = self._sourceReactor
+        refAssem = src.core.refAssem
+        refNumPoints = (
+            len(src.core.findAllAxialMeshPoints([refAssem], applySubMesh=True)) - 1
+        )  # pop off zero
+        # skip the first value of the mesh (0.0)
+        allMeshes = []
+        for a in src.core:
+            aMesh = src.core.findAllAxialMeshPoints([a], applySubMesh=True)[1:]
+            if len(aMesh) == refNumPoints:
```

Comment:
this is a  pretty big assumption and deserves a comment.  What if there are multiple assems that have same numb of points?

Reply:
i know its moved code, but maybe just add a comment saying 
# If lengths are the same, this assembly contributes to the uniform mesh

In the future it would be good to confirm that the meshes detected this way are all roughly similar.

Reply:
Right. I'll add a comment to this because this was just what has been there for a _long_ time. I think the point of this is that we want to make sure if there is some burn-up or different thermal expansion for different fuel assemblies that we average out those changes rather than just assuming the refAssem mesh.

Reply:
Added a detailed comment.

Reply:
I like the note you added, but right above the if statement is where i was hoping to see a comment, because that if statement is saying 
"if the lenghs are equal, it must be part of the uniform mesh i am trying to create", but what if it just happens to be equal

Reply:
Okay I'll update again